### ISSUE TITLE
[BUG][csharp-netcore] Seems like optionalProjectFile=false isn't used with csharp-netcore

### ISSUE THREAD

**Issue by Cyber1000** [OpenAPITools/openapi-generator#14460]
##### Description

- Executing the following: 
`openapi-generator-cli generate -i https://petstore3.swagger.io/api/v3/openapi.json -g csharp-netcore -o api --additional-properties=optionalProjectFile=false,sourceFolder=test`
- Same Problem with master: tested with latest master (amd64 - digest on dockerhub: 267dafcbfb97):
`docker run --rm -v "${PWD}:/local" openapitools/openapi-generator-cli:latest generate \
    -i https://petstore3.swagger.io/api/v3/openapi.json -g csharp-netcore -o api \ --additional-properties=optionalProjectFile=false,sourceFolder=test \
    -o /local/out/csharp`
- I found optionalProjectFile only in readmes like this, but not in code: https://github.com/OpenAPITools/openapi-generator/blob/master/docs/generators/csharp-netcore.md
- expected behaviour: at the end of generating der is no Org.OpenAPITools.csproj
- actual behaviour: this flag doesn't change anything, as far as I can see

##### openapi-generator version

- 6.2.1 (but even master)

##### Related issues/PRs

- No related found so far

##### Suggest a fix

- did search through the organisation here with https://github.com/search?p=4&q=org%3AOpenAPITools+optionalProjectFile&type=Code
   - found ~ 30 Code-sources, looked through it which of them belong to csharp-netcore and only found Readme-Files and similar. Code usages only in java-generators.
- No fix so far, but if someone can point me to the right file I may help here

# Additional question
- should a solution file get created if I don't want to create a projectFile? I would not expect
<img width="160" alt="image" src="https://user-images.githubusercontent.com/1142765/212491453-675edab6-65e8-4ca1-a370-03bf49f2038f.png">

**Comment by Cyber1000** [comment]
Found this file which seems to use optionalProjectFileFlag (excluding sln + csproj if false), I just don't know what I'm doing wrong here.

**Comment by Cyber1000** [comment]
Sorry didn't mention the file in my latest post: found the usage in https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CSharpClientCodegen.java but not in https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CSharpNetCoreClientCodegen.java


So -g csharp works as expected, but -g csharp-netcore doesn't.

**Comment by Cyber1000** [comment]
btw. there seems to be an undocumented additionalparameter named "excludeTests" whcih seems to work for both templates

**Comment by vicdotexe** [comment]
Any update on this? I'm using the latest version where csharp-netcore got merged with csharp, so I'm just using `-g csharp`, but `optionalProjectFile=false` still seems to produce a csproj file.

**Comment by aong3-wx** [comment]
Yep, getting the same problem, where `optionalProjectFile=false` inside `config.json` still produces a csproj file.

**Comment by jonferreira** [comment]
2026 and still no fix?
